pkg/server/chaosd/attack.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chaosd
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/pingcap/log"
+	perr "github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/chaos-mesh/chaosd/pkg/core"
+)
+
+type Environment struct {
+	AttackUid string
+	Chaos     *Server
+}
+
+type AttackType interface {
+	Attack(options core.AttackConfig, env Environment) error
+	Recover(experiment core.Experiment, env Environment) error
+}
+
+func (s *Server) newEnvironment(uid string) Environment {
+	return Environment{
+		AttackUid: uid,
+		Chaos:     s,
+	}
+}
+
+func (s *Server) ExecuteAttack(attackType AttackType, options core.AttackConfig) (uid string, err error) {
```

Comment:
LGTM. I think add some comment for this important function is better. It may include how the attacks are scheduled.

Reply:
Agreed, added a docstring